picard/ui/itemviews.py

```diff
@@ -1160,47 +1168,41 @@ def update(self, update_track=True, update_selection=True):
             self.setText(i, file.column(column[1]))
             self.setForeground(i, color)
             self.setBackground(i, bgcolor)
-        if file.errors:
-            self.setToolTip(MainPanel.TITLE_COLUMN, _("Processing error(s): See the Errors tab in the File Info dialog"))
         if update_selection and self.isSelected():
             TreeItem.window.update_selection(new_selection=False)
         parent = self.parent()
         if isinstance(parent, TrackItem) and update_track:
             parent.update(update_files=False, update_selection=update_selection)
 
     @staticmethod
-    def decide_file_icon(file):
+    def decide_file_icon_info(file):
+        tooltip = ""
         if file.state == File.ERROR:
             if file.error_type == FileErrorType.NOTFOUND:
-                return FileItem.icon_error_not_found
+                icon = FileItem.icon_error_not_found
+                tooltip = _("File not found")
             elif file.error_type == FileErrorType.NOACCESS:
-                return FileItem.icon_error_no_access
+                icon = FileItem.icon_error_no_access
+                tooltip = _("File permission error")
             else:
-                return FileItem.icon_error
+                icon = FileItem.icon_error
+                tooltip = _("Processing error(s): See the Errors tab in the File Info dialog")
         elif isinstance(file.parent, Track):
             if file.state == File.NORMAL:
-                return FileItem.icon_saved
+                icon = FileItem.icon_saved
+                tooltip = _("Track saved")
             elif file.state == File.PENDING:
-                return FileItem.match_pending_icons[int(file.similarity * 5 + 0.5)]
+                icon = FileItem.match_pending_icons[int(file.similarity * 5 + 0.5)]
+                tooltip = _("Pending")
             else:
-                return FileItem.match_icons[int(file.similarity * 5 + 0.5)]
+                icon = FileItem.match_icons[int(file.similarity * 5 + 0.5)]
+                tooltip = _(FileItem.match_icons_info[int(file.similarity * 5 + 0.5)])
```

Comment:
This calculation is repeated 3 times, maybe we could move it to a separate function.

Also the matching is a bit weird, as a `similarity` of 0 selects the icon 50.

Similarity, index, icon:

```
0.00 0  50
0.05 0  50
0.10 1  60
0.15 1  60
0.20 1  60
0.25 1  60
0.30 2  70
0.35 2  70
0.40 2  70
0.45 2  70
0.50 3  80
0.55 3  80
0.60 3  80
0.65 3  80
0.70 4  90
0.75 4  90
0.80 4  90
0.85 4  90
0.90 5 100
0.95 5 100
1.00 5 100
```

Reply:
@zas I addressed the code duplication in d753c69f690b69cfe32f5f7801cb04f299f5b995

> Also the matching is a bit weird, as a similarity of 0 selects the icon 50.

That's because we have 5 different quality match icons, with ` match-50.png`  being the lowest. The icons are not a direct visualization of the matching score. The score itself has very little meaning in itself anyway, it's mostly useful when being compared with other scores.

I wouldn't change this really. I don't think we would gain much from introducing for example 5 additional icons with even smaller bars. For one those small bars become essentially indistinguishably (at some point even the color does not matter anymore), but also for existing users the icons would just start to appear as worse matches then in previous Picard versions.